src/cfn/template.yaml

```
@@ -592,3 +618,37 @@ Resources:
       MemorySize: 128
       Timeout: 30
       Role: !GetAtt CustomResourceRole.Arn
+  CloudFrontOriginAccessIdentity:
+    Type: AWS::CloudFront::CloudFrontOriginAccessIdentity
+    Condition: WithCloudFront
+    Properties:
+      CloudFrontOriginAccessIdentityConfig:
+        Comment: !Ref WebUIBucket
+  CloudFrontDistribution:
+    Type: AWS::CloudFront::Distribution
+    Condition: WithCloudFront
+    Properties:       
+      DistributionConfig:
+        Origins:
+        - DomainName: !GetAtt WebUIBucket.RegionalDomainName
+          Id: !Sub myS3Origin-${CollectionId}
+          S3OriginConfig:
+            OriginAccessIdentity: !Sub origin-access-identity/cloudfront/${CloudFrontOriginAccessIdentity}
+        Enabled: true
+        HttpVersion: http2
+        Comment: The Distribution for the Rekognition Meter Web UI
+        DefaultRootObject: index.html
+        DefaultCacheBehavior:
+          AllowedMethods:
+          - HEAD
+          - GET
+          - OPTIONS
+          TargetOriginId: !Sub myS3Origin-${CollectionId}
+          ForwardedValues:
+            QueryString: false
+            Cookies:
+              Forward: none
+          ViewerProtocolPolicy: allow-all
```

```suggestion
          ViewerProtocolPolicy: redirect-to-https
```
Unless there's a reason to provide HTTP access, lets redirect to HTTPS by default?